# Теория вероятностей и математическая статистика
## Урок 3. EDA (exploratory data analysis) или Разведочный анализ

In [1]:
import numpy as np
from math import factorial

In [2]:
def combinations(n, k):
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))

**Задача 1**  
Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [3]:
salary = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [4]:
# Среднее арифметическое
salary_mean = salary.sum() / salary.shape[0]
salary_mean

65.3

In [5]:
# Смещенная дисперсия
biased_var = ((salary - salary_mean)**2).sum() / (salary.shape[0])
print(biased_var)
# СКО
st_dev_biased = np.sqrt(biased_var)
print(round(st_dev_biased, 2))

950.11
30.82


In [6]:
# Смещенная дисперсия
unbiased_var = ((salary - salary_mean)**2).sum() / (salary.shape[0] - 1)
print(round(unbiased_var, 2))
# СКО
st_dev_unbiased = np.sqrt(unbiased_var)
print(round(st_dev_unbiased, 2))

1000.12
31.62


----------------------

**Ответ:**  
1) *Среднее арифметическое* = 65.3  
2) *Смещенная дисперсия* = 950.11, *СКО* = 30.82  
3) *Несмещенная дисперсия* = 1000.12, *СКО* = 31.62  

--------------------------

**Задача 2**  
В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [7]:
# Из 1-го ящика достали 0 белых, из второго - 3 белых
p1 = ((combinations(5,0) * combinations(3,2)) / combinations(8,2)) * ((combinations(5,3) * combinations(7,1)) / combinations(12,4))

# Из 1-го ящика достали 1 белый, из второго - 2 белых
p2 = ((combinations(5,1) * combinations(3,1)) / combinations(8,2)) * ((combinations(5,2) * combinations(7,2)) / combinations(12,4))

# Из 1-го ящика достали 2 белых, из второго - 1 белый
p3 = ((combinations(5,2) * combinations(3,0)) / combinations(8,2)) * ((combinations(5,1) * combinations(7,3)) / combinations(12,4))

In [8]:
# Какова вероятность того, что 3 мяча белые?
round(p1+p2+p3, 2)

0.37

------------------

**Ответ:**  0.37

---------------------

**Задача 3:**  
На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.  

A - попадание в мишень  
B1 - стрелял 1 спортсмен (=1/3)  
B2 - стрелял 2 спортсмен  (=1/3)  
B3 - стрелял 3 спортсмен  (=1/3)  

In [9]:
# Общая вероятность
general_prob = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6

In [10]:
# Стрелял 1 спортсмен
P1 = (0.9 * 1/3) / general_prob
round(P1, 2)

0.39

In [11]:
# Стрелял 2 спортсмен
P2 = (0.8 * 1/3) / general_prob
round(P2, 2)

0.35

In [12]:
# Стрелял 3 спортсмен
P3 = (0.6 * 1/3) / general_prob
round(P3, 2)

0.26

-----------------

**Ответ:**  
а) *Вероятность того, что выстрел произведен первым спортсменом* = 0.39  
б) *Вероятность того, что выстрел произведен вторым спортсменом* = 0.35  
в) *Вероятность того, что выстрел произведен третьим спортсменом* = 0.26

-----------------------------

**Задача 4:**  
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

A - сессия сдана  

A + B + C = 1

B1 - студент факультета А (=1/4)  
B2 - студент факультета B  (=1/4)  
B3 - студент факульетта С  (=2/4=1/2)  

In [13]:
# Общая вероятность
general_prob = 1/4 * 0.8 + 1/4 * 0.7 + 1/2 * 0.9

In [14]:
# студент факультета А
P1 = (0.8 * 1/4) / general_prob
round(P1, 2)

0.24

In [15]:
# студент факультета B
P2 = (0.7 * 1/4) / general_prob
round(P2, 2)

0.21

In [16]:
# студент факультета C
P3 = (0.9 * 1/2) / general_prob
round(P3, 2)

0.55

----------------

**Ответ:**  
а) *Студент учится на факультете А* = 0.24    
б) *Студент учится на факультете В* = 0.21  
в) *Студент учится на факультете С* = 0.55  

--------------------

**Задача 5**  
Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [17]:
# Выйдут из строя все детали
p1 = 0.1 * 0.2 * 0.25
round(p1, 4)

0.005

In [18]:
# Выйдут из строя 2 детали 
#   3 не вышла из строя     1 не вышла       2  не вышла
p2 = 0.1 * 0.2 * 0.75 + 0.9 * 0.2 * 0.25 + 0.1 * 0.8 * 0.25
round(p2, 4)

0.08

In [19]:
# Выйдет из строя хотя бы 1 деталь = 1 - не выйдет ни одна деталь
p3 = 1 - 0.9 * 0.8 * 0.75
round(p3, 3)

0.46

In [20]:
# Выйдет от 1 до 2 деталей = 1 деталь + 2 детали
p4 = p2 + 0.1 * 0.8 * 0.75 + 0.9 * 0.2 * 0.75 + 0.9 * 0.8 * 0.25
round(p4, 3)

0.455

-------------

**Ответ:**  
а) *Все детали выйдут из строя* = 0.005  
б) *2 детали выйдут из строя* = 0.08  
в) *Хотя бы 1 деталь выйдет из строя* = 0.46  
г) *От 1 до 2 деталей выйдут из строя* = 0.455